src/vulkan/tlas.cc

```diff
@@ -0,0 +1,251 @@
+// Copyright 2024 The Amber Authors
+// Copyright (C) 2024 Advanced Micro Devices, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "src/vulkan/tlas.h"
+#include "src/vulkan/blas.h"
+
+namespace amber {
+namespace vulkan {
+
+const VkTransformMatrixKHR identityMatrix3x4 = {{{1.0f, 0.0f, 0.0f, 0.0f},
+                                                 {0.0f, 1.0f, 0.0f, 0.0f},
+                                                 {0.0f, 0.0f, 1.0f, 0.0f}}};
+
+VkTransformMatrixKHR makeVkMatrix(const float* m) {
+  VkTransformMatrixKHR v;
+
+  if (m == nullptr)
+    return identityMatrix3x4;
+
+  for (size_t i = 0; i < 12; i++) {
+    const size_t r = i / 4;
+    const size_t c = i % 4;
+    v.matrix[r][c] = m[i];
+  }
+
+  return v;
+}
+
+TLAS::TLAS(Device* device) : device_(device) {}
+
+Result TLAS::CreateTLAS(amber::TLAS* tlas, blases_map* blases, tlases_map* tlases) {
+  if (tlas_ != VK_NULL_HANDLE)
+    return {};
+
+  assert(tlas != nullptr);
+
+  VkDeviceOrHostAddressConstKHR constNullPtr;
+  VkDeviceOrHostAddressKHR nullPtr;
```

Comment:
Can we rename these from nullPtr to something like `const_default_ptr` and `default_ptr`? It's really hard to read the `nullptr` vs `nullPtr` when scanning through the code.

Reply:
ok